pkg/daemon/update.go

```diff
@@ -48,6 +59,64 @@ func (dn *Daemon) update(oldConfig, newConfig *mcfgv1.MachineConfig) error {
 	return dn.reboot()
 }
 
+// reconcilable checks the configs to make sure that the only changes requested
+// are ones we know how to do in-place. if we can't do it in place, the node is
+// marked as degraded.
+//
+// we can only update machine configs that have changes to the files,
+// directories, links, and systemd units sections of the included ignition
+// config currently.
+func (dn *Daemon) reconcilable(oldConfig, newConfig *mcfgv1.MachineConfig) (bool, error) {
+	oldIgn := oldConfig.Spec.Config
+	newIgn := newConfig.Spec.Config
+
+	// Ignition section
+
+	// if the config versions are different, all bets are off. this probably
+	// shouldn't happen, but if it does, we can't deal with it.
+	if oldIgn.Ignition.Version != newIgn.Ignition.Version {
+		return false, nil
```

Comment:
Should there be an log message that occurs here? I haven't looked in all calling functions but I could see confusion if the versions don't match, the admin/user doesn't know how to inspect the versions directly, and is told the old and new configs are not reconcilable.

Reply:
Same question for following similar areas.

Reply:
Yeah, logging the specific thing that went wrong is definitely a good idea.